### ISSUE TITLE
Some outpost npcs and AI crew really love stairs in city outposts

### ISSUE THREAD

**Issue by Hamburgian** [FakeFishGames/Barotrauma#12888]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I noticed this a really long time ago, however I assumed it would have gotten fixed by now, but it hasn't so I am here reporting this small bug. In **city** outposts, some of the special Npcs (and crew members too, rarely) will decide to run to the far bottom right of the city outpost and run back and forth on a staircase. This mainly happens with the outpost manager but only after they've given you your missions.

https://github.com/Regalis11/Barotrauma/assets/141864587/cfade70d-1636-4ca2-ab57-767e3bd3502c

### Reproduction steps

1. dock at city outpost
2. speak to mission man
3. speak to him again and again until they have given you all their missions
4. observe as start to run away
5. "hey where are you going man???"
6. follow them
7. they end up at the bottom right side of the outpost running in circles around a set of stairs

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Single player

### Version

v1.1.18.1 (Treacherous Tides)

### Which operating system did you encounter this bug on?

Windows

**Comment by itchyOwl** [comment]
Thanks for a detailed bug report and an illustrative video about it. However, I don't think there's a bug there. The bots don't have knowledge about the automatic mechanics. And since there's no buttons connected to the door that they could interact to open it, they think they don't have access to it. Instead of just running into the door, they stop, as intended. In this particular case it's not the optimal behavior, but I don't think we should change the AI logic in trying to "fix" it. Instead, you could add buttons to the door for overriding the automatic mechanic, so that the bots could use them for opening the door.

**Comment by Hamburgian** [comment]
this report doesn't involve the door mechanics though. Are you saying that if you mess with the doors so that they can't get there whatsoever then it would be solved or is that something else?

**Comment by Regalis11** [comment]
@itchyOwl did you mean to reply to [AI won't walk through custom Autodoor](https://github.com/Regalis11/Barotrauma/issues/12889)?

**Comment by itchyOwl** [comment]
Yep. Not sure what happened there. Sorry :D

**Comment by itchyOwl** [comment]
This issue could be a waypoint thing.

**Comment by marpul358** [comment]
Station administrator has captain job behaviour and it is trying to get to a navigation console according to Discord discussion.

**Comment by itchyOwl** [comment]
> Station administrator has captain job behaviour and it is trying to get to a navigation console according to Discord discussion.

It's ok for them to have a captain job, but ofc they shouldn't try navigating with the outpost :D If you suspect, that's the case, please type debugai in the console and take a screenshot. Couldn't repro it.

**Comment by Jasontti** [comment]
Tested in: https://github.com/FakeFishGames/Barotrauma-development/pull/5164/commits/1757a5b33a73013ea47057f06dee1dc986ae4ee1

Tested with different behaviours of npcs, follow command, natural walking, going for objective, running.
They don't stop now in the middle of the stairs now, which was the original issue.

New issue when npc is walking to a destination, like outpost director or outpost doctor going back after being teleported, they can be sucked in to the stairs and get stuck.
Sometimes they can get out of it by themselves. Usually security guard with changing destination endpoint, but some are permanently stuck.

https://github.com/FakeFishGames/Barotrauma/assets/8488224/95a286bb-60c2-4476-ab18-925aa7ce6248

**Comment by itchyOwl** [comment]
Fixed the code causing it in FakeFishGames/Barotrauma-development@9098742. Also adjusted the waypoints a bit.

**Comment by Jasontti** [comment]
Tested in: https://github.com/FakeFishGames/Barotrauma-development/pull/5164/commits/b637c3fdfbcd6f3f055ef8447da9950e4db872fe

Tested the issue case above
Observed how NPCs go about in the stairs in SP and MP

No issues found
Closing ticket
